src/modules/network/components/networkSwitcherDropdown/NetworkSwitcherDropdown.js

```diff
@@ -62,29 +65,24 @@ function NetworkSwitcherDropdown({ noLabel, onLoadApplications, onLoadNetworkSta
 
       setApplications(defaultApplications);
     }
-
-    onLoadApplications?.({
-      isErrorGettingApplication: blockchainAppsMetadata.isError,
-      isGettingApplication: blockchainAppsMetadata.isLoading,
-      applications: defaultApplications,
-    });
   }, [
     blockchainAppsMetadata.isLoading,
     blockchainAppsMetadata.isError,
     blockchainAppsMetadata.isFetched,
   ]);
 
   useEffect(() => {
-    onLoadNetworkStatus?.({
-      isGettingNetworkStatus: networkStatus.isLoading,
-      isErrorGettingNetworkStatus: networkStatus.isError,
-      selectedNetworkStatusData: networkStatus.data?.data,
-    });
-  }, [networkStatus.isLoading, networkStatus.isError]);
-
-  useEffect(() => {
-    blockchainAppsMetadata.refetch();
-  }, [selectedNetwork]);
+    const isSuccess =
+      blockchainAppsMetadata.isSuccess &&
+      networkStatus.isSuccess &&
+      !blockchainAppsMetadata.isFetching &&
+      !networkStatus.isFetching;
+    if (isSuccess) {
+      onNetworkSwitchSuccess?.(true);
+    } else {
+      onNetworkSwitchSuccess?.(false);
+    }
```

Comment:
this can be 

`onNetworkSwitchSuccess?.(isSuccess);`

Reply:
Nice, it will even be more KIS :D